File: app/src/main/java/io/github/golok56/manajemenuang/ui/user/newuser/NewUserViewModel.java
package io.github.golok56.manajemenuang.ui.user.newuser;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;

import io.github.golok56.manajemenuang.R;
import io.github.golok56.manajemenuang.utility.PreferenceManager;

/**
 * @author Satria Adi Putra
 */

public class NewUserViewModel extends AndroidViewModel {
    private PreferenceManager mPref;

    public NewUserViewModel(Application app) {
        super(app);
        mPref = PreferenceManager.getInstance(app);
    }

    public void createUser(String name, String nominal) {
        mPref.setUserName(name);
        mPref.setWalet(nominal);
    }

    public String getValidationText() {
        return getApplication().getString(R.string.validasi_dialog);
    }

}

File: app/src/main/java/io/github/golok56/manajemenuang/viewmodels/NewUserViewModel.kt
package io.github.golok56.manajemenuang.viewmodels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel

import io.github.golok56.manajemenuang.R
import io.github.golok56.manajemenuang.utility.PreferenceManager

/**
 * @author Satria Adi Putra
 */

class NewUserViewModel(app: Application) : AndroidViewModel(app) {
    private val mPref: PreferenceManager = PreferenceManager.getInstance(app)

    val validationText: String
        get() = getApplication<Application>().getString(R.string.validasi_dialog)

    fun createUser(name: String, nominal: String) {
        mPref.userName = name
        mPref.setWalet(nominal)
    }

}
